platform/security/certificate/security-certificate-generator/src/main/java/org/codice/ddf/security/certificate/generator/CertificateCommand.java

```diff
@@ -18,42 +18,83 @@
-
     /**
      * Pass in a string to use as the common name of the certificate to be generated.
-     * Exception thrown if 0 arguments or more than 1 argument.
+     * Exception thrown if 0 arguments or more than 2 argument (or more than 4 when -san is used).
+     * <pre>
+     * Arguments to this program are: <code>[-cn &lt;cn&gt;|-dn &lt;dn&gt;] [-san &lt;tag:name,tag:name,...&gt;]</code>
      *
-     * @param args
+     * where:
+     * &lt;cn&gt; represents a fully qualified common name (e.g. "&lt;FQDN&gt;", where &lt;FQDN&gt; could be something like cluster.yoyo.com)
+     * &lt;dn&gt; represents a distinguished name as a comma-delimited string (e.g. "c=US, st=California, o=Yoyodyne, l=San Narciso, cn=&lt;FQDN&gt;")
+     * &lt;tag:name,tag:name,...&gt; represents optional subject alternative names to be added to the generated certificate
+     *    (e.g. "DNS:&lt;FQDN&gt;,DNS:node1.&lt;FQDN&gt;,DNS:node2.&lt;FQDN&gt;"). The format for subject alternative names
+     *    is similar to the OpenSSL X509 configuration format. Supported tags are:
+     *      email - email subject
+     *      URI - uniformed resource identifier
+     *      RID - registered id
+     *      DNS - hostname
+     *      IP - ip address (V4 or V6)
+     *      dirName - directory name
+     * </pre>
+     *
+     * @param args arguments to the certificate program (see above for description)
      */
     public static void main(String args[]) {
-        if (args.length != 2) {
-            String canonicalName = CertificateCommand.class.getCanonicalName();
-            String exCn = String.format("java %s -cn \"John Smith\"", canonicalName);
+        // try to extract -san if provided
+        int expected = 2;
+        String[] sans = null;
+        int cnDnPosition = 0;
 
-            String exDn = String.format(
-                    "java %s -dn \"cn=John Whorfin, o=Yoyodyne, l=San Narciso, st=California, c=US\"",
-                    canonicalName);
-            String usage = String.format(
-                    "%nUsage: java %s [-cn <common name>] | [-dn <distinguished name>]%n Examples: %n%s%n%s",
-                    canonicalName, exCn, exDn);
-            throw new RuntimeException(usage);
+        if (args.length == 4) {
+            if (args[0].trim()
+                    .equalsIgnoreCase("-san")) {
+                sans = Arrays.stream(args[1].split("[,]"))
+                        .map(String::trim)
+                        .toArray(String[]::new);
+                cnDnPosition = 2;
+                expected = 4;
+            } else if (args[2].trim()
+                    .equalsIgnoreCase("-san")) {
+                sans = Arrays.stream(args[3].split("[,]"))
+                        .map(String::trim)
+                        .toArray(String[]::new);
+                expected = 4;
+            }
         }
-
-        if (args[0].trim()
+        if (args.length != expected) {
+            throw new RuntimeException(String.format(
```

Comment:
That is the original code and I didn't want to change the original behavior unless you think we should. I would never throw RuntimeException on my own 😉